go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go

```diff
@@ -74,26 +75,59 @@ const (
 
 func buildPlayerPlan(filter *binlogdatapb.Filter, tableKeys map[string][]string) (*PlayerPlan, error) {
 	plan := &PlayerPlan{
-		VStreamFilter: &binlogdatapb.Filter{
-			Rules: make([]*binlogdatapb.Rule, len(filter.Rules)),
-		},
-		TablePlans: make(map[string]*TablePlan),
+		VStreamFilter: &binlogdatapb.Filter{},
+		TargetTables:  make(map[string]*TablePlan),
+		TablePlans:    make(map[string]*TablePlan),
 	}
-	for i, rule := range filter.Rules {
-		if strings.HasPrefix(rule.Match, "/") {
-			plan.VStreamFilter.Rules[i] = rule
-			continue
-		}
-		sendRule, tablePlan, err := buildTablePlan(rule, tableKeys)
-		if err != nil {
-			return nil, err
+	for tableName := range tableKeys {
+		for _, rule := range filter.Rules {
+			switch {
+			case strings.HasPrefix(rule.Match, "/"):
+				expr := strings.Trim(rule.Match, "/")
```

Comment:
The explanation for this change could be derived from the description in the PR. But this part of the code is tricky. I think starting to document in detail what's going on here, will be very helpful for our future selves.

Reply:
Added a high level comment for buildPlayerPlan.